api/control/v1alpha1/types.go

```diff
@@ -257,3 +257,46 @@ type BindingList struct {
 	metav1.ListMeta `json:"metadata,omitempty"`
 	Items           []Binding `json:"items"`
 }
+
+// CustomClean describes how to clean a resource.
+// That is, how to clean objects of a given resource.
+// The name of this object is the `resource.apiGroup` of the resource
+// that it is about; for example, `appwrappers.workload.codeflare.dev`.
+// The cleaning specified here is in addition to whatever cleaning
+// is built into KubeStellar for that resource.
+//
+// +genclient
+// +genclient:nonNamespaced
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:scope=Cluster,shortName={cc},categories={all}
+type CustomClean struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec CustomCleanSpec `json:"spec,omitempty"`
+}
+
+// CustomCleanSpec describes how to clean an object.
+type CustomCleanSpec struct {
+
+	// `remove` is a list of JSONPath expressions (https://goessner.net/articles/JsonPath/)
+	// that identify part of the object to remove if present.
+	// Only a subset of JSONPath is supported.
+	// The expression used in a filter must be a conjunction of field == literal tests.
+	// Examples:
+	// - "$.spec.resources.GenericItems[*].generictemplate.metadata.resourceVersion"
+	// - "$.store.book[?(@.author == 'Kilgore Trout' && @.category == 'fiction'].price"
+	Remove []string `json:"remove,omitempty"`
```

Comment:
I guess if we only "clean" then "remove is OK, but I would expect a general json patch string here

Reply:
I also think we may do remove/update/add. 
it would be good to have kind of `operator` (add/update/remove).